Adding share target results in Gradle exception

On Bubblewrap 1.8.2, is share target really supported? I see it in `TwaManifest.shareTarget`, but if I try to use it, it breaks.

Specifically, if I create a `TwaManifest` with the following `shareTarget`, Bubblewrap will fail to generate the package.

```typescript
const manifest: TwaManifest = {
   "host": "mathbymiles.com",
   "shareTarget": {
         "action":"/new-topic",
         "method":"GET",
         "enctype":"application/x-www-form-urlencoded",
         "params": {
              "title":"title",
              "text":"body"
          }
   },
   ...omitted for brevity
}
```

If I try to build the package, I receive the following error:

> Details: Error generating app package: Error: Command failed: ./gradlew assembleRelease --stacktrace
> FAILURE: Build failed with an exception.
>
> What went wrong:
> Execution failed for task ':app:generateReleaseBuildConfig'.
> Manifest file does not exist: /tmp/pwabuilder-cloudapk--36-B2nqEkzP1ZR3/app/src/main/AndroidManifest.xml

This error was originally reported to PWABuilder [here](https://github.com/pwa-builder/PWABuilder/issues/1386). Others are experiencing the same issue with different URLs, e.g. https://github.com/pwa-builder/PWABuilder/issues/1215#issuecomment-758593296.

--- a/packages/core/src/lib/TwaGenerator.ts
+++ b/packages/core/src/lib/TwaGenerator.ts
@@ -347,7 +347,7 @@ export class TwaGenerator {
     if (twaManifest.shareTarget?.params?.files) {
       shareTargetIntentFilter.actions.push('android.intent.action.SEND_MULTIPLE');
       for (const file of twaManifest.shareTarget.params.files) {
-        file.accept.forEach((accept) => shareTargetIntentFilter!.mimeTypes.push(accept));
+        file.accept.forEach((accept) => shareTargetIntentFilter.mimeTypes.push(accept));
       }
     } else {
       shareTargetIntentFilter.mimeTypes.push('text/plain');

--- a/packages/core/src/lib/TwaGenerator.ts
+++ b/packages/core/src/lib/TwaGenerator.ts
@@ -102,6 +102,11 @@ const NOTIFICATION_IMAGES: IconDefinition[] = [
 const WEB_MANIFEST_LOCATION = '/app/src/main/res/raw/';
 const WEB_MANIFEST_FILE_NAME = 'web_app_manifest.json';
 
+type ShareTargetIntentFilter = {
+  actions: string[];
+  mimeTypes: string[];
+};
+
 function shortcutMaskableTemplateFileMap(assetName: string): Record<string, string> {
   return {
     'app/src/main/res/drawable-anydpi-v26/shortcut_maskable.xml':
@@ -328,6 +333,27 @@ export class TwaGenerator {
     }));
   }
 
+  private static generateShareTargetIntentFilter(
+      twaManifest: TwaManifest): ShareTargetIntentFilter | undefined {
+    if (!twaManifest.shareTarget) {
+      return undefined;
+    }
+
+    const shareTargetIntentFilter: ShareTargetIntentFilter = {
+      actions: ['android.intent.action.SEND'],
+      mimeTypes: [],
+    };
+
+    if (twaManifest.shareTarget?.params?.files) {
+      shareTargetIntentFilter.actions.push('android.intent.action.SEND_MULTIPLE');
+      for (const file of twaManifest.shareTarget.params.files) {
+        file.accept.forEach((accept) => shareTargetIntentFilter!.mimeTypes.push(accept));
+      }
+    } else {
+      shareTargetIntentFilter.mimeTypes.push('text/plain');
+    }
+    return shareTargetIntentFilter;
+  }
   /**
    * Creates a new TWA Project.
    *
@@ -363,6 +389,7 @@ export class TwaGenerator {
     const args = {
       ...twaManifest,
       ...features,
+      shareTargetIntentFilter: TwaGenerator.generateShareTargetIntentFilter(twaManifest),
       generateShortcuts: twaManifest.generateShortcuts,
       escapeJsonString: escapeJsonString,
       toAndroidScreenOrientation: toAndroidScreenOrientation,

--- a/packages/core/src/lib/TwaManifest.ts
+++ b/packages/core/src/lib/TwaManifest.ts
@@ -321,13 +321,15 @@ export class TwaManifest {
 
   private static verifyShareTarget(
       webManifestUrl: URL, shareTarget?: ShareTarget): ShareTarget | undefined {
-    if (!shareTarget?.action || !shareTarget?.params?.files) {
+    if (!shareTarget?.action) {
       return undefined;
     }
 
-    for (const file of shareTarget.params.files) {
-      if (!file.accept) {
-        return undefined;
+    if (shareTarget?.params?.files) {
+      for (const file of shareTarget.params.files) {
+        if (!file.accept) {
+          return undefined;
+        }
       }
     }
 

--- a/packages/core/template_project/app/src/main/AndroidManifest.xml
+++ b/packages/core/template_project/app/src/main/AndroidManifest.xml
@@ -124,12 +124,22 @@
             <% if (shareTarget) { %>
                 <intent-filter>
                     <action android:name="android.intent.action.SEND" />
-                    <action android:name="android.intent.action.SEND_MULTIPLE" />
+                    <% 
+                       // SEND_MULTIPLE is added when handling images, videos, etc.  
+                       if (shareTarget.params.files) { %>
+                        <action android:name="android.intent.action.SEND_MULTIPLE" />
+                    <% } %>
                     <category android:name="android.intent.category.DEFAULT" />
-                    <% for (const file of shareTarget.params.files) { %>
-                        <% for (const accept of file.accept) { %>
-                            <data android:mimeType="<%= accept %>" />
+                    <% 
+                       if (shareTarget.params.files) { %>
+                        <% for (const file of shareTarget.params.files) { %>
+                            <% for (const accept of file.accept) { %>
+                                <data android:mimeType="<%= accept %>" />
+                            <% } %>
                         <% } %>
+                    <% } else {%>
+                        // When params.files is not available, use "text/plain" the mime-type.  
+                        <data android:mimeType="text/plain" />
                     <% } %>
                 </intent-filter>
             <% } %>

--- a/packages/core/template_project/app/src/main/AndroidManifest.xml
+++ b/packages/core/template_project/app/src/main/AndroidManifest.xml
@@ -123,13 +123,13 @@
 
             <% if (shareTargetIntentFilter) { %>
                 <intent-filter>
-                    <% for (const action of shareTargetIntentFilter.actions) %>
+                    <% for (const action of shareTargetIntentFilter.actions) { %>
                         <action android:name="<%= action %>" />
-                    <% 
+                    <% } %>
                     <category android:name="android.intent.category.DEFAULT" />
-                    <% for (const mimeType of shareTargetIntentFilter.mimeTypes) %>
+                    <% for (const mimeType of shareTargetIntentFilter.mimeTypes) { %>
                         <data android:mimeType="<%= mimeType %>" />
-                    <% 
+                    <% } %>
                 </intent-filter>
             <% } %>
 

--- a/packages/core/template_project/app/src/main/AndroidManifest.xml
+++ b/packages/core/template_project/app/src/main/AndroidManifest.xml
@@ -121,26 +121,15 @@
             <meta-data android:name="android.support.customtabs.trusted.SCREEN_ORIENTATION"
                 android:value="@string/orientation"/>
 
-            <% if (shareTarget) { %>
+            <% if (shareTargetIntentFilter) { %>
                 <intent-filter>
-                    <action android:name="android.intent.action.SEND" />
+                    <% for (const action of shareTargetIntentFilter.actions) %>
+                        <action android:name="<%= action %>" />
                     <% 
-                       // SEND_MULTIPLE is added when handling images, videos, etc.  
-                       if (shareTarget.params.files) { %>
-                        <action android:name="android.intent.action.SEND_MULTIPLE" />
-                    <% } %>
                     <category android:name="android.intent.category.DEFAULT" />
+                    <% for (const mimeType of shareTargetIntentFilter.mimeTypes) %>
+                        <data android:mimeType="<%= mimeType %>" />
                     <% 
-                       if (shareTarget.params.files) { %>
-                        <% for (const file of shareTarget.params.files) { %>
-                            <% for (const accept of file.accept) { %>
-                                <data android:mimeType="<%= accept %>" />
-                            <% } %>
-                        <% } %>
-                    <% } else {%>
-                        // When params.files is not available, use "text/plain" the mime-type.  
-                        <data android:mimeType="text/plain" />
-                    <% } %>
                 </intent-filter>
             <% } %>
 

